### ISSUE TITLE
Kernel Panic upgrading from Monterey 12.0.1 to 12.02

### ISSUE THREAD

**Issue by enigma0456** [myspaghetti/macos-virtualbox#578]
Have successfully upgraded from the original Catalina through Big Sur and to Monterey until this. VirtualBox with 2 cpus and 8 gb ram allocated. Continuous kernel panics when trying this latest upgrade on two different machines and hosts. Reading comments here I reduced the machine to 1 cpu instead of two and although a little loggy, the upgrade ran through without incident. After optimization, powered down and rebooted with two processors. At the sign on screen there was a kernel panic and it rebooted. I let it and after a little bit of spinning beach ball, it came up and seems to be functional. System info indicates 8gb ram and 2 cores as expected. No request for fix - just providing data for analysis. Thanks for your efforts!

**Comment by jimbolimbo13** [comment]
Thanks for sharing. I've been unable to upgrade from Big Sur to Monterey. It keeps ending in a kernel panic boot loop. Currently using 4 cores and 8gb ram. I'll try it with 1cpu and see if it works 🙏

**Comment by beriaanirudh** [comment]
Thanks for sharing this @enigma0456 @jimbolimbo13 . I upgraded to Monterey, it looped in boot, I reduced to 1 CPU and upgraded, which succeeded. Now, with 2 CPUs, kernel panics as soon as I open a terminal / Xcode. It works with 1 CPU, so I am not blocked, but its slow. Can you please share other VM configs that you are using? Specifically these are mine:

1. Motherboard: Base Memory **8192MB**
2. Motherboard: Chipset: **ICH9** 
3. Motherboard: Pointing Device **USB Tablet**
4. Motherboard: Enable I/O APIC (**true**)
5. Motherboard: Enable EFI (**true**)
6. Motherboard: Hardware Clock in UTC time (**true**)
7. Processor: Execution cap **100%**
8. Processor: Enable PAE/NX (**true**)
9. Processor: Enable Nested VT-x/AMD-V (**false**)
10. Acceleration: Enable Nested Paging (**false**)


Can you please point what config/s are different in your setup? PS: I am on Ubuntu 20.04 on AMD Ryzen 5

**Comment by enigma0456** [comment]
The only setting that is different is the last - I have nested paging enabled. Running Ubuntu 21.10 on HP Probook 4440s. I basically ran the script to create the Catalina machine with all the extra settings and have successfully upgraded through all the Big Sur iterations and then on into Monterey. Monterey has proven to be less stable. I have used VB for many years on Linux hosts running Windows VMs in production. I have always seen issues when you allocate more than one processor to the guest. so this doesn't surprised me. My thinking is that even though the guest has only one processor the host can use multiple cores to service the guest requests. I'm no expert in this so take it with a grain of salt. I have found that making a VB clone of the original machine and then using that as a base for fresh installs, upgrades, etc has been very useful. Make new UUIDs and Mac Addresses when you clone to avoid VB complaints about duplicates. Fun stuff courtesy of myspaghetti!!!
